src/backend/deep_search/views.py

```diff
@@ -0,0 +1,283 @@
+"""
+Albert chatbot integration views.
+
+This module provides the core API endpoints for the chatbot functionality.
+"""
+
+import logging
+from typing import Dict, Any
+
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework import status
+
+from .chatbot import get_chatbot
+from .email_service import EmailService
+
+logger = logging.getLogger(__name__)
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def intelligent_search_api(request):
+    """
+    Intelligent email search endpoint using Albert API.
+    
+    This endpoint analyzes the user's first 500 emails and returns
+    the best matches using AI-powered search.
+    
+    POST data:
+    {
+        "query": "User's natural language search query"
+    }
+    
+    Returns:
+    {
+        "success": true,
+        "results": [...],
+        "search_summary": "...",
+        "total_matches": 5,
+        "total_emails": 500,
+        "query": "original query"
+    }
+    """
+    try:
+        data = request.data
+        query = data.get('query', '').strip()
+        
+        if not query:
+            return Response({
+                'success': False,
+                'error': 'query is required'
+            }, status=status.HTTP_400_BAD_REQUEST)
+        
+        # Get user ID from authenticated user
+        if not hasattr(request, 'user') or not request.user.is_authenticated:
+            return Response({
+                'success': False,
+                'error': 'Authentication required'
+            }, status=status.HTTP_401_UNAUTHORIZED)
+        
+        user_id = str(request.user.id)
+        logger.info(f"Processing intelligent search for user {user_id}: {query}")
+        
+        # Use actual email service with RAG search
+        try:
+            # Initialize email service
+            email_service = EmailService()
+            
+            # Get chatbot instance for API client
+            chatbot = get_chatbot()
+            
+            # Perform intelligent email search using RAG
+            search_results = email_service.chatbot_intelligent_email_search(
+                user_id=user_id,
+                user_query=query,
+                api_client=chatbot.api_client,
+                max_results=10
```

Comment:
Could we imagine to let api consumer to set this property through a query param? Or is it a threshold for performance purpose?